backend/src/models/reports/Tab_44_rpt_PF_RecoveryToDateDetails.js

```diff
@@ -1,16 +1,151 @@
-//const dbConnection = require("@database/databaseConnection");
-//const { knex } = dbConnection();
+// libs
+const { knex } = require("@database/databaseConnection")();
 
-const getReport = (portfolio) => {
-  const query = "removeme"; //knex().raw();
-  return query;
+// fist CTE
+
+const baseQueries = {
+  q1: `(
+  SELECT c.project_id,
+    c.fiscal,
+    sum(c.total_fee_amount) AS fees,
+    sum(c.total_expense_amount) AS expenses,
+    sum(total_fee_amount + total_expense_amount) AS total_contract
+  FROM contract c
+  GROUP BY c.project_id,
+    c.fiscal
+  HAVING c.project_id IS NOT NULL)`,
+
+  q2: `(
+  SELECT p.project_number,
+    p.project_name,
+    p.total_project_budget,
+    q1.total_contract,
+    p.recoverable_amount,
+    sum(q1_amount + q2_amount + q3_amount + q4_amount) AS current_fy_total_recoverable,
+    sum(
+      CASE WHEN q1_recovered THEN q1_amount ELSE cast(0 AS MONEY) END +
+      CASE WHEN q2_recovered THEN q2_amount ELSE cast(0 AS MONEY) END +
+      CASE WHEN q3_recovered THEN q3_amount ELSE cast(0 AS MONEY) END +
+      CASE WHEN q4_recovered THEN q4_amount ELSE cast(0 AS MONEY) END
+    ) AS current_fy_recovered_to_date,
+    fy.id as fiscal_year
+  FROM project p
+    RIGHT JOIN (
+      (
+        (
+          fiscal_year fy
+          RIGHT JOIN project_deliverable pd ON fy.id = pd.fiscal
+        )
+        LEFT JOIN q1 ON (pd.fiscal = q1.fiscal)
+        AND (pd.project_id = q1.project_id)
+      )
+      RIGHT JOIN (
+        portfolio po
+        RIGHT JOIN project_budget pb ON po.id = pb.recovery_area
+      ) ON pd.id = pb.project_deliverable_id
+    ) ON p.id = pd.project_id
+  GROUP BY p.id,
+    p.project_number,
+    p.project_name,
+    po.portfolio_name,
+    po.portfolio_abbrev,
+    p.total_project_budget,
+    p.recoverable_amount,
+    q1.fees,
+    q1.expenses,
+    q1.total_contract,
+    pd.fiscal,
+    fy.fiscal_year,
+    fy.id
+    )`,
+};
+
+/**
+ * Retrieves the data for various financial metrics based on the fiscal year.
+ *
+ * @param   {number | string | Array} Parameter- The fiscal, Date, or Portfolio(s) to grab data for
+ * @returns {Promise}                            - A promise that resolves to the query result
+ */
+const reportQueries = {
+  fiscal: ({ fiscal }) =>
+    knex("fiscal_year").select("fiscal_year").where("fiscal_year.id", fiscal).first(),
+
+  main: knex.raw(
+    `
+    (
+      WITH
+      q1 AS ${baseQueries.q1},
```

Comment:
@kshapka-bcgov I'm curious what you think about this way of breaking out subqueries

it looks like it takes about 127 ms to run on the backend, so it's pretty light

Reply:
Could be a decent compromise on readability vs effort. Of course, this is a pretty simple subset of base queries, so I wonder if it will still be a performance issue on more bloated queries...

Reply:
That's an interesting question. I suppose we'll learn more on future reports. 

I'm guessing that there's a point of diminishing returns as far as decomposing a query into parcels of knex() methods, and that this is pretty much as good as performance gets, without an SQL refactor, but If I'm wrong, I'll put in more effort to wrap things up.

so far, I've worked two bug tickets where bloated queries take around 100 ms in PgAdmin, but 5,000+ (intermittently) in knex() wrapped versions. I'm still not confident I know what specifically causes this, but it seems to get fixed when I reduce the amount of knex per SQL